tools/resourcedocsgen/cmd/metadata.go

```diff
@@ -191,6 +192,9 @@ func packageMetadataFromGitHubCmd(metadataDir, packageDocsDir *string) *cobra.Co
 			"be inferred to be provider/cmd/pulumi-resource-<providerName>/schema.json")
 	cmd.Flags().StringVar(&publisher, "publisher", "", "The publisher's display name to be shown in the package. "+
 		"This will default to Pulumi")
+	cmd.Flags().StringVar(&editURLAllowList, "editURLAllowList", "", "A comma-separated list of URLs that are allowed "+
+		"to have an edit_url in the frontmatter. If not specified, no edit_url will be added to the frontmatter. "+
+		"This is used to prevent the edit_url from being added to the frontmatter by default.")
```

Comment:
I don't understand why we need this. Who will manually set this flag. This CLI is almost always called in CI, not by hand.

In my mind, we have 3 situations:
1. We pull a file from a remote source (where we insert the `"# WARNING: this file was fetched from ..." comment). 
  a. If that's a force (GitHub, GitLab, etc.), we know the edit URL - it can be derived from the source URL.
  b. The URL is from static storage (like S3). There is no edit URL.
3. We don't have a remote source, so the best place to edit is the file in GH. In that case, we should keep the existing edit URL.

Can we put those situations in the code.